utils/rowgroup/rowgroup.h

```diff
@@ -984,17 +1021,30 @@ inline void Row::colUpdateHasherTypeless(datatypes::MariaDBHasher& h, uint32_t k
   }
 }
 
+inline void Row::setStringField(const uint8_t* str, uint32_t length, uint32_t colIndex)
+{
+  utils::ConstString temp((const char*)str, length);
+  setStringField(temp, colIndex);
+}
+inline void Row::setStringField(const utils::NullString& val, uint32_t colIndex)
+{
+  utils::ConstString temp(val.str(), val.length());
+  setStringField(temp, colIndex);
+}
 inline void Row::setStringField(const utils::ConstString& str, uint32_t colIndex)
 {
   uint64_t offset;
 
   // TODO: add multi-byte safe truncation here
   uint32_t length = str.length();
-  if (length > getColumnWidth(colIndex))
-    length = getColumnWidth(colIndex);
+
+  setNullMark(colIndex, !str.str());
 
   if (inStringTable(colIndex))
   {
+    if (length > getColumnWidth(colIndex))
```

Comment:
There is a common if here and at line 1056. Plz move it back and put outside if that starts at 1043.

Reply:
Done.